ssl/record/rec_layer_s3.c

```diff
@@ -938,6 +938,7 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
         }
 
         if (SSL_TREAT_AS_TLS13(s)
+                && !BIO_get_ktls_send(s->wbio)
```

Comment:
Probably we not go down the KTLS path if `record_padding_cb` is set. Also if the max_fragment_length extension is in use then ssl_get_max_send_fragment() could return something smaller than normal - so we probably shouldn't use KTLS in that scenario either (I wonder if we have considered this in the TLSv1.2 case?)

Reply:
Yes, that is checked on configuration stage